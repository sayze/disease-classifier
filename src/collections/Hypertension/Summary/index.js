import React from 'react'
import { Typography, Grid } from '@material-ui/core'
import PropTypes from 'prop-types'

const Summary = ({ date, sysBp, dialBp }) => (
  <Grid container direction="column" spacing={1}>
    <Grid item>
      <Typography variant="body1" component="p">
        Reading Date: {date}
      </Typography>
    </Grid>
    <Grid item>
      <Typography variant="body1" component="p">
        Systolic Blood Pressure: {sysBp}
      </Typography>
    </Grid>
    <Grid item>
      <Typography variant="body1" component="p">
        Diastolic Blood Pressure: {dialBp}
      </Typography>
    </Grid>
  </Grid>
)

Summary.defaultProps = {
  date: new Date().toDateString(),
  sysBp: 0,
  dialBp: 0,
}

Summary.propTypes = {
  date: PropTypes.string,
  sysBp: PropTypes.number,
  dialBp: PropTypes.number,
}

export default Summary
