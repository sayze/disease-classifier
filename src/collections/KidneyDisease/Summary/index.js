import React from 'react'
import { Typography, Grid } from '@material-ui/core'
import PropTypes from 'prop-types'
import dateWithFormat from 'services/Date'

const Summary = ({ date, eGFR, percentage }) => (
  <Grid container direction="column" spacing={1}>
    <Grid item>
      <Typography variant="body1" component="p">
        Reading Date: {dateWithFormat(date)}
      </Typography>
    </Grid>
    <Grid item>
      <Typography variant="body1" component="p">
        eGFR: {eGFR}
      </Typography>
    </Grid>
    <Grid item>
      <Typography variant="body1" component="p">
        Percent Change: {percentage}%
      </Typography>
    </Grid>
  </Grid>
)

Summary.defaultProps = {
  date: new Date(),
  eGFR: 0,
  percentage: 0,
}

Summary.propTypes = {
  date: PropTypes.object,
  eGFR: PropTypes.number,
  percentage: PropTypes.number,
}

export default Summary
