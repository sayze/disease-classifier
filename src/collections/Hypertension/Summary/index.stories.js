import React from 'react'
import Summary from '.'
import { withKnobs, number } from '@storybook/addon-knobs'

export default {
  title: 'Hypertension Summary',
  component: Summary,
  decorators: [withKnobs],
}
export const Default = () => {
  const dialBp = number('Diastolic Blood Pressure', 0)
  const sysBp = number('Systolic Blood Pressure', 0)
  return <Summary date="21/05/2020" dialBp={dialBp} sysBp={sysBp} />
}
