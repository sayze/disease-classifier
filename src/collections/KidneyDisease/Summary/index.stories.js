import React from 'react'
import Summary from '.'
import { withKnobs, number } from '@storybook/addon-knobs'

export default {
  title: 'Kidney Disease Summary',
  component: Summary,
  decorators: [withKnobs],
}
export const Default = () => {
  const eGFR = number('eGFR', 0)
  const percentage = number('Percentage', 0)
  return <Summary eGFR={eGFR} percentage={percentage} />
}
