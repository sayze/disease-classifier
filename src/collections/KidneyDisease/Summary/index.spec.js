import React from 'react'
import { shallow } from 'enzyme'
import Summary from '.'

describe('Kidney Disease Summary', () => {
  it('should render a valid kidney disease summary', () => {
    const component = shallow(<Summary date={new Date('2020-01-01')} />)
    expect(component).toMatchSnapshot()
  })
})
