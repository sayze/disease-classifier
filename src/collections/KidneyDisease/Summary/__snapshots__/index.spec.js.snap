// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kidney Disease Summary should render a valid kidney disease summary 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    >
      Reading Date: 
      2020/01/01
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    >
      eGFR: 
      0
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    >
      Percent Change: 
      0
      %
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
